### PR TITLE
MB-16029, MB-16381 - Edit sit in any status

### PR DESCRIPTION
**Pull request by michaelvaldes-truss**
## Summary

This is the second PR created for the story, with commits from the first PR (https://github.com/transcom/mymove/pull/10640) included in this branch.
This PR attempts to allow the user to edit the SIT status of any shipment that is currently in SIT, or _will be_ in SIT that will start in the future. It also attempts to correct the various date and count values displayed in the SIT dashboard and edit modals. Finally, it aims to hide irrelevant current/future SIT fields as per MB-16381.

Testing for this PR should be made easier with the inclusion of testharness scenarios that demonstrate SIT edge cases, including: 

- HHGMoveIn200DaysSITEndsToday
- HHGMoveIn200DaysSITEndsTomorrow
- HHGMoveIn200DaysSITEndsYesterday
- HHGMoveIn200DaysSITDeparted

One thing to give extra attention: the database is storing certain SIT dates without time(zone) information, so the front end and swagger were updated to reflect that.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. Access the testharness via the url [baseurl]/testharness/list
2. Create a move from one of the scenarios listed higher up in this PR description
3. Copy the move code from the generated data 
4. Access the office app as a TOO
5. Navigate to the move corresponding to the copied move code
6. Navigate to the MTO tab
7. Test the SIT display and edit modals, picking from the below criteria depending on the scenario being tested

Some things to test when looking at the SIT dashboard for a particular move:
- the allowance is editable for both current and future SIT
- the values for the SIT day counts (e.g. "Total days used") make sense and add up correctly
- the values for the current/future SIT dates (e.g. "SIT authorized end date) make sense and add up correctly
- the values for the current/future SIT dates and Edit button are hidden when there is no current/future SIT

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/about/supported-browsers) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/about/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

## Screenshots
Shipment with ongoing SIT
image

Shipment with SIT that ends "today" (as of the snapping of this screenshot)
image

Shipment with SIT that has remained in storage, exceeding its allowance
image

Shipment with SIT that has departed from storage before using all of its allowance
image

### CODE REVIEW

**Comment by AnnaGingle** [comment]
I am seeing an issue with the authorized end date & the modal.

https://github.com/transcom/mymove/assets/48488699/a031d2c0-b1dc-40a9-98c8-b381e8b08f05

**Comment by AnnaGingle** [comment]
Following up on my previous comment, the modal saves the changes when using the  Total days of SIT approved field.

https://github.com/transcom/mymove/assets/48488699/6ad4b50f-f1f3-439b-a1c1-eff7da2f8dd1

File: swagger-def/definitions/SITStatus.yaml [hunk 1]
```diff
@@ -14,10 +14,15 @@ properties:
...
+    format: date
+    x-nullable: true
   sitDepartureDate:
     type: string
-    format: date-time
+    format: date
+    x-nullable: true
+  sitAllowanceEndDate:
+    type: string
+    format: date
```
Comment (rogeruiz) [inline swagger-def/definitions/SITStatus.yaml:25]: Nice! I see we're reverting from `2017-07-21T17:32:28Z` to `2017-07-21` for the SIT dates.

**Review by rogeruiz — COMMENTED** [review]
> Only commenting for now. I will check out the calculations on Monday. If this gets approved by @AnnaGingle before then, I will also approve. The code looks correct, but I'd like to ensure the functionality matches my understanding of the code.

**Review by AnnaGingle — APPROVED** [review]
Great work! The Modal is working as expected.

**Review by rogeruiz — APPROVED** [review]
lgtm 🌈

code looks great. Thanks for refactoring this and digging into the API responses to match how dates are stored in the DB for SIT.

**Pull request merged**